e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1518,3 +1518,20 @@ func RootSyncHasNomosStatus(expectedCommit, expectedStatus string) Predicate {
 		return nil
 	}
 }
+
+// ConfigMapHasData returns an error if the ConfigMap doesn't contain the given key value pair
+func ConfigMapHasData(key string, value string) Predicate {
+	return func(o client.Object) error {
+		if o == nil {
+			return ErrObjectNotFound
+		}
+		cm, ok := o.(*corev1.ConfigMap)
+		if !ok {
+			return WrongTypeErr(o, cm)
+		}
+		if cm.Data[key] == value {
```

Comment:
Suggest checking that the value exists as well. Currently this does not differentiate between the key missing and the key containing an empty string.

```golang
if gotVal, ok := cm.Data[key]; !ok {
  return fmt.Errorf(...)
} else if gotVal != value {
  return fmt.Errorf(...)
}
return nil
```

Reply:
Thanks, done